### PR TITLE
Cool new explorer widget

### PR DESCRIPTION
**Pull request by ericdill**
cc @danielballan @tacaswell

### CODE REVIEW

**Comment by danielballan** [comment]
Yeah, this looks good.

**Comment by danielballan** [comment]
I'll let @tacaswell sign off in case there are mpl subtleties I don't see.

**Comment by ericdill** [comment]
How should I test this thing? I guess just poke the traits that back thr widgets and make sure they respond correctly...?  Is there any guidelines for CI testing of ipywidgets?

**Comment by danielballan** [comment]
Yeah, that's what I would do.

File: databroker/widgets/explorer.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    key_select.on_trait_change(scan_submit, 'value')
+
+    widgets = {'x_dropdown': x_dropdown,
+               'y_select': y_select,
+               'scan_select': scan_select,
+               'fill_checkbox': fill_checkbox,
+               'key_select': key_select,
+               'scan_text': scan_text}
+    box = VBox([fill_checkbox, HBox([scan_text, key_select]), scan_select, x_dropdown, y_select])
+    return namedtuple('explorer_return_values', ['widgets', 'display'])(widgets, box)
```
Comment (tacaswell) [inline databroker/widgets/explorer.py]: Probably worth creating the named tuple once rather than every time.

**Comment by tacaswell** [comment]
Sorry, started to review this and ended up down a rabbit hole which may result in a PR to pandas....

**Comment by danielballan** [comment]
I am taking the "WIP" commit to mean, "stand by, this isn't quite ready more review." Tag me when this is ready. :- )

**Comment by danielballan** [comment]
Let's get this in. What remains to be done, @ericdill?

**Comment by ericdill** [comment]
There are no tests on the backend logic of the widget.  That being said, I'm not going to be able to write said tests for a while.  There are other, higher priority issues that need to be dealt with (like vertical integration testing!)  Let me rebase it (the conflict is likely in the meta.yaml) and then make sure it still runs.  Have you tried to run it on your local machine?

**Pull request closed without merging**